### PR TITLE
add whitelist urls to env file

### PR DESCRIPTION
**Pull request by Arjun99ab**
previously, on a self hosted instance users would receive a cors error when fetching resources from their hosted instance.

now, there is a clear area in the .env to add whitelist urls which cors.js handles

### CODE REVIEW

**Comment by amosyu2000** [comment]
@Arjun99ab , thanks for doing this, will review and merge when I get a chance.

File: .env [hunk 1]
```diff
@@ -5,7 +5,9 @@
 # Your Hypixel secret key, see https://developer.hypixel.net/
 HYPIXEL_SECRET_KEY=
 
-
+# URLs of your 25karma frontend (whitelist)
+# Format: "[url] [url] ..." (space-separated). e.g "https://25karma.xyz http://localhost:3000"
+FRONTEND_URLS=""
```
Comment (amosyu2000) [inline .env]: Can we call this `WHITELISTED_URLS` instead? We can also make it an optional variable - if no URLs are provided then allow all URLs through.
```suggestion
WHITELISTED_URLS=
```

File: middlewares/cors.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import expressCors from 'cors';
 
-const whitelist = [];
+const whitelist = process.env.FRONTEND_URLS.split(' ')
```
Comment (amosyu2000) [inline middlewares/cors.js]:
```suggestion
const SEPARATOR = ' ';
const whitelist = process.env.WHITELISTED_URLS;
```

Also you can update line 7 to be
```
origin: whitelist !== undefined ? whitelist.split(SEPARATOR) : '*',
```

**Review by amosyu2000 — CHANGES_REQUESTED** [review]
Some small changes requested, then it's good to merge.

File: .env (hunk 1 above)
Comment (Arjun99ab) [reply .env]: Yeah that's better

File: middlewares/cors.js (hunk 2 above)
Comment (Arjun99ab) [reply middlewares/cors.js]: I see, yeah that looks better

**Review by amosyu2000 — APPROVED** [review]
lgtm

**Pull request merged**